### ISSUE TITLE
Pyston is slower on some benchmarks significantly in comparison with other languages like C#, Perl, Go, etc.

### ISSUE THREAD

**Issue by ghost** [pyston/pyston_v2#188]
This came out of discussion:
https://github.com/pyston/pyston/issues/129

The benchmark is here:
https://github.com/pepe78/python-performance/

More details in here:
https://github.com/pepe78/python-performance/issues/1

**Comment by kmod** [comment]
Yeah choosing a language for a project is an interesting problem, and we're hoping to make Python a bit more attractive by making it a bit faster. That said, we're not the ones picking Python's overall position in the PL universe, for that you might want to head over to [python-dev](https://mail.python.org/mailman3/lists/python-dev.python.org/). There does seem to be a bit of a trend to migrate workloads from Python to faster languages like Go and Rust, so you're not wrong about the tradeoffs.

Our goal is to make application code faster, which looks different than the microbenchmark you wrote. And the fact that we're 60% faster on it without specifically optimizing it feels like a positive outcome to me. I guess you want us to be even faster? We could do that, and it'd make our microbenchmark numbers better, but would be at the expense of the code that our users actually care about.

If you haven't already I'd recommend checking out numpy, which lets you offload most of your computation out of Python -- I believe that this is the solution that most researchers use, because it lets them get the power of Python while not paying the performance penalty for the bulk of the work.

**Comment by ghost** [comment]
I can close this if you want... I only opened it because you asked me to :)

I know about numpy.
